clayui.com/src/components/Editor/index.js

```diff
@@ -38,39 +36,31 @@ const Editor = ({
 	preview = true,
 	scope = {},
 }) => {
-	let defaultSnippet = {
-		code: '',
-		disabled: false,
-		format: true,
-		imports,
-		name: 'React',
-	};
-
 	if (Array.isArray(code)) {
-		const codeToFormat = [...code];
-
-		codeToFormat.forEach((snippet) => {
-			if (snippet.format) {
-				snippet.code = formatCode(snippet.code);
-			}
+		code = code.map(({code, format = true, ...other}) => {
+			return {
+				...other,
+				code: format ? formatCode(code) : code,
+			};
 		});
-
-		code = codeToFormat;
-
-		defaultSnippet = code[REACT_SNIPPET_INDEX];
 	} else {
-		code = formatCode(code);
-
-		defaultSnippet.code = code;
+		code = [
+			{
+				disabled: false,
+				imports,
+				name: 'React',
+				value: formatCode(code),
+			},
+		];
```

Comment:
Simplified this so that `code` always ends up being an array. This way we can simplify some logic below and now have to check if its an array or not